src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -41,8 +42,13 @@ export type TopNavMenuProps = StatefulSearchBarProps & {
  *
  **/
 
-export function TopNavMenu(props: TopNavMenuProps) {
-  const { config, showSearchBar, ...searchBarProps } = props;
+export function TopNavMenu(props: TopNavMenuProps): ReactElement | null {
+  const { config, showTopNavMenu, showSearchBar, ...searchBarProps } = props;
+
+  if (!showTopNavMenu && (!showSearchBar || !props.data)) {
```

Comment:
Ideally, we wouldn't even be rendering this component is `showTopNavMenu` is false.

Reply:
I could rename it to `showMenu`, as I think it's still useful to pick if the menu and search bar are rendered separately. However, I'll take another look at using `config` instead, and moving the conditional to the dashboard when getting the menu items to show.